test_wazuh/test_fim/test_skip/test_skip.py

```diff
@@ -0,0 +1,99 @@
+                                                   {'FIM_MODE': '', 'SKIP': 'no'},
+                                                   {'FIM_MODE': {'realtime': 'yes'}, 'SKIP': 'yes'},
+                                                   {'FIM_MODE': {'realtime': 'yes'}, 'SKIP': 'no'},
+                                                   {'FIM_MODE': {'whodata': 'yes'}, 'SKIP': 'yes'},
+                                                   {'FIM_MODE': {'whodata': 'yes'}, 'SKIP': 'no'}
+                                                   ],
+                                           metadata=[{'fim_mode': 'scheduled', 'skip': 'yes'},
+                                                     {'fim_mode': 'scheduled', 'skip': 'no'},
+                                                     {'fim_mode': 'realtime', 'skip': 'yes'},
+                                                     {'fim_mode': 'realtime', 'skip': 'no'},
+                                                     {'fim_mode': 'whodata', 'skip': 'yes'},
+                                                     {'fim_mode': 'whodata', 'skip': 'no'}
+                                                     ]
+                                           )
+
+
+# fixtures
+
+@pytest.fixture(scope='module', params=configurations)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope='session')
+def configure_nfs():
+    # create and configure nfs mount point
+    subprocess.call(['./data/configure_nfs.sh'])
+    yield
+    # remove nfs
+    subprocess.call(['./data/remove_nfs.sh'])
+    shutil.rmtree(os.path.join('/', 'media', 'nfs-folder'), ignore_errors=True)
+
+
+# tests
+
+
+@pytest.mark.parametrize('directory,  tags_to_apply', [
+    (os.path.join('/', 'proc'), {'skip_proc'}),
+    (os.path.join('/', 'sys'), {'skip_sys'}),
+    (os.path.join('/', 'dev'), {'skip_dev'}),
+    (os.path.join('/', 'nfs-mount-point'), {'skip_nfs'})
+])
+def test_skip(directory, tags_to_apply,
+              get_configuration, configure_environment, configure_nfs,
+              restart_syscheckd, wait_for_initial_scan):
+    """ Check if syscheck is skipping the directory based on its skip configuration
+
+    :param directory: Directory that will be monitored
+    """
+    check_apply_test(tags_to_apply, get_configuration['tags'])
+    timeout = 3
+    if tags_to_apply == {'skip_proc'}:
+        timeout = 100
+
+    if get_configuration['metadata']['skip'] == 'yes':
+        trigger = False
+    else:
+        trigger = True
+
+    if tags_to_apply == {'skip_proc'} or tags_to_apply == {'skip_sys'}:
+        if trigger:
+            event = wazuh_log_monitor.start(timeout=8, callback=callback_detect_integrity_event).result()
+            assert any(path in event['data'].get('path') for path in ['/proc', '/sys'])
+        else:
+            with pytest.raises(TimeoutError):
```

Comment:
I suggest fetching an explicit message in the log confirming the event has been skipped instead of waiting a TimeoutError to be raised. Otherwise, the test could be too slow.

Reply:
`skip_proc` and `skip_sys` tests were modified since they were not accurate enough. In my opinion, waiting for a TimeoutError is safer because we assert there are not events and it is a 3 seconds wait for 2 tests.